packages/pluggableWidgets/accordion-web/src/components/Accordion.tsx

```diff
@@ -23,9 +37,39 @@ export function Accordion(props: AccordionProps): ReactElement | null {
 
     const [collapsedAccordionGroups, collapsedAccordionGroupsDispatch] = useReducer(
         reducer.current,
-        props.groups.map(() => !props.previewMode && props.collapsible)
+        props.groups.reduce<boolean[]>((accumulator, group) => {
```

Comment:
Is this one a candidate for `flatMap`?

Reply:
I think this cannot be used, because we need to change previous values in the array. See line 45.

Reply:
As discussed it will be changed to map and an if outside.

Reply:
Optimized it.